languages/csharp/exercises/concept/expression-bodied-members/.meta/Example.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Collections.Generic;
+
+public class WeatherStation
+{
+    private Reading reading;
+    private List<DateTime> recordDates = new List<DateTime>();
+    private List<decimal> temperatures = new List<decimal>();
+
+    public int AcceptReading(Reading reading)
+    {
+        this.reading = reading;
+        recordDates.Add(DateTime.Now);
+        temperatures.Add(reading.Temperature);
+        return recordDates.Count;
+    }
+
+    public void ClearAll()
+    {
+        this.reading = new Reading();
+        recordDates.Clear();
+        temperatures.Clear();
+    }
+
+    public decimal LatestTemperature => reading.Temperature;
+
+    public decimal LatestPressure => reading.Pressure;
+
+    public decimal LatestRainfall => reading.Rainfall;
+
+    public bool HasHistory => recordDates.Count > 1;
+
+    public Outlook Outlook
+        => reading.Equals(new Reading()) ? throw new ArgumentException()
+            : reading.Pressure < 10m && reading.Temperature < 30m ? Outlook.Cool
+            : reading.Temperature > 50 ? Outlook.Good
+            : Outlook.Warm;
```

Comment:
I'm not actually sure I like this version more than the original one :D But I'm generally not in favor of chaining ternary expressions. For education purposes, this can work though.

Reply:
I agree that ternary expressions get confusing quickly.  However, if every other member in the class is an expression-bodied member I would make an exception.  Another approach (though not here, as we are demonstrating compound ternaries) is to use a switch expression, relying on the case guards and ignoring the pattern.  It seems a little abusive but I would accept it.